src/functional/iterator/transforms/cse.py

```diff
@@ -0,0 +1,52 @@
+from collections import Counter
+
+from eve import NodeTranslator, NodeVisitor
+from eve.utils import UIDs
+from functional.iterator import ir
+
+
+class CollectSubexpressions(NodeVisitor):
```

Comment:
How about a docstring with an example (doctest if possible), similar to the passes in field view? This is not a prototype anymore.